### ISSUE TITLE
esp-idf upgrade to v5.3?

### ISSUE THREAD

**Issue by xiaguangbo** [esp-rs/esp-idf-template#228]
v5.3 realese

**Comment by xiaguangbo** [comment]
change `ESP_IDF_VERSION = "v5.3"`, do`cargo clean` `cargo build` is ok

**Comment by SergioGasquez** [comment]
Hi! This update is already on the way with https://github.com/esp-rs/esp-idf-template/pull/226

**Comment by Vollbrecht** [comment]
Yeah we need some adjustment in the HAL and new releases of it. Till then you can try to use the p4-support branch of the hals. I try to come back to it and finish it soon.

**Comment by xiaguangbo** [comment]
@SergioGasquez @Vollbrecht thanks!

**Comment by ivmarkov** [comment]
Fixed.
